### PR TITLE
Feature/backend update patient

### PR DESCRIPTION
**Pull request by andreibalau**
Implementation for patient endpoint (PUT request)

- to be able to update a patient with a new doctor

### CODE REVIEW

File: backend/SCP-Interns-Project-Backend/src/main/java/com/cerner/SCPInternsProjectBackend/controller/PatientApi.java [hunk 1]
```diff
@@ -36,19 +48,32 @@
...
+                    System.out.println("plm");
+                    return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+                }
+            }
         } else {
             log.warn("ObjectMapper or HttpServletRequest not configured in default PatientApi interface so no example is generated");
         }
+        System.out.println("cool");
         return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
     }
```
Comment (imartinas) [inline backend/SCP-Interns-Project-Backend/src/main/java/com/cerner/SCPInternsProjectBackend/controller/PatientApi.java]: Please remove the write to console lines.

File: backend/SCP-Interns-Project-Backend/src/main/java/com/cerner/SCPInternsProjectBackend/controller/PatientApiController.java [hunk 2]
```diff
@@ -34,6 +34,7 @@
 	
 		
 			try {
+				System.out.println(servicePatient.updatePatient(patientId,body));
```
Comment (imartinas) [inline backend/SCP-Interns-Project-Backend/src/main/java/com/cerner/SCPInternsProjectBackend/controller/PatientApiController.java]: Please remove the write to console.

File: backend/SCP-Interns-Project-Backend/src/main/java/com/cerner/SCPInternsProjectBackend/service/PatientService.java [hunk 3]
```diff
@@ -22,23 +24,18 @@ private File getPatientsFile() throws FileNotFoundException {
...
-		System.out.println(this.findPatientById());
-//		System.out.println(patientId);
-//		System.out.println(body);
-//		System.out.println(this.getPatientsFile());
-		
-		return null;
+
+		PatientDto patient = this.findPatientById(patientId);
+		patient.setDoctor(body);
+//		System.out.println(patient);
```
Comment (imartinas) [inline backend/SCP-Interns-Project-Backend/src/main/java/com/cerner/SCPInternsProjectBackend/service/PatientService.java]: Remove commented line.

File: backend/SCP-Interns-Project-Backend/src/main/java/com/cerner/SCPInternsProjectBackend/controller/PatientApi.java [hunk 4]
```diff
@@ -58,18 +58,22 @@
...
+                    System.out.println("plm");
                     return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
                 }
             }
         } else {
             log.warn("ObjectMapper or HttpServletRequest not configured in default PatientApi interface so no example is generated");
         }
+        System.out.println("cool");
         return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
     }
```
Comment (imartinas) [inline backend/SCP-Interns-Project-Backend/src/main/java/com/cerner/SCPInternsProjectBackend/controller/PatientApi.java]: Please remove the write to console lines.

**Review by imartinas — CHANGES_REQUESTED** [review]

**Review by imartinas — APPROVED** [review]

**Review by andreeacojocaru731 — APPROVED** [review]

**Pull request merged**